capz/run-capz-e2e.sh

```diff
@@ -28,11 +28,13 @@ main() {
     export CAPI_EXTENSION_SOURCE="${CAPI_EXTENSION_SOURCE:-"https://github.com/Azure/azure-capi-cli-extension/releases/download/v0.1.3/capi-0.1.3-py2.py3-none-any.whl"}"
     export IMAGE_SKU="${IMAGE_SKU:-"${WINDOWS_SERVER_VERSION:=windows-2019}-containerd-gen1"}"
     
-    # CI is an environment variable set by a prow job: https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables
+    # CI and JOB_TYPE areenvironment variables set by the prow job: https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables
     export CI="${CI:-""}"
+    export JOB_TYPE="${JOB_TYPE:-""}"
 
     set_azure_envs
     set_ci_version
+    if [[ "${JOB_TYPE}" == "presubmit" ]]; then "${CAPZ_DIR}/scripts/ci-build-kubernetes.sh"; fi
```

Comment:
do you think we need `&& [[ "${REPO_OWNER:-}/${REPO_NAME:-}" == "kubernetes/kubernetes" ]];`?

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e1e1a1270b663a9b256f7d41783603f529454549/hack/util.sh#L38

Reply:
@CecileRobertMichon suggested to just use https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/hack/util.sh#L32